src/import/FSHImporter.ts

```diff
@@ -1498,19 +1501,26 @@ export class FSHImporter extends FSHVisitor {
     return caretValueRule;
   }
 
-  visitCodeCaretValueRule(ctx: pc.CodeCaretValueRuleContext): CodeCaretValueRule {
-    const codePath = ctx.CODE().map(code => {
-      return this.parseCodeLexeme(code.getText(), ctx).code;
-    });
-    const codeCaretValueRule = new CodeCaretValueRule(codePath)
+  visitCodeCaretValueRule(ctx: pc.CodeCaretValueRuleContext): CaretValueRule | CodeCaretValueRule {
+    const localCodePath = ctx.CODE()
+      ? ctx.CODE().map(code => {
+          return this.parseCodeLexeme(code.getText(), ctx).code;
+        })
+      : [];
+    const fullCodePath = this.getCodePathWithContext(localCodePath, ctx);
+    // If there is a code path, use it to make a CodeCaretValueRule.
+    // Otherwise, make a CaretValueRule.
+    const caretRule = (
+      fullCodePath.length > 0 ? new CodeCaretValueRule(fullCodePath) : new CaretValueRule('')
+    )
```

Comment:
I can't comment on the actual line, but in the `CodeSystemExporter`, in `setCodeCareRules`, we don't do `resolveSoftIndexing`. I think I would expect something like: 
```
CodeSystem: MyCS
* #foo
  * #bar
    * #bam
    * ^designation[+].value = "index 0"
    * ^designation[+].value = "index 1"
```
to work, considering something like:
```
CodeSystem: MyCS
* ^identifier[+].value = "foo"
* ^identifier[+].value = "bar"
```
does work.

Reply:
I'm also not sure why we need `CaretValueRules` at all. Can't a `CodeCaretValueRule` represent everything that a `CaretValueRule` can, just by having an empty `codePath`? On the export side, it looks like the processing is almost exactly the same.

Reply:
Huh, somehow I was under the impression that soft indexing was not allowed in caret paths, which is not true at all. So yes, since soft indexing is allowed on caret paths in `CaretValueRule`, it should also be allowed on caret paths in `CodeCaretValueRule`.

Reply:
And, I think that `CaretValueRule` still needs to be allowed on a CodeSystem so that inserted RuleSets can still provide them, but on the import side, I think it should be fine to always provide a `CodeCaretValueRule`. I'll try it out and see if anything bad happens.